packages/cardano-services/src/ChainHistory/DbSyncChainHistory/mappers.ts

```diff
@@ -105,14 +105,29 @@ export const mapWithdrawal = (withdrawalModel: WithdrawalModel): Cardano.Withdra
 // Remove this and select the actual redeemer data from `redeemer_data` table.
 const stubRedeemerData = Buffer.from('not implemented');
 
+const redeemerPurposeMap: Record<RedeemerModel['purpose'], Cardano.RedeemerPurpose> = {
+  cert: Cardano.RedeemerPurpose.certificate,
+  mint: Cardano.RedeemerPurpose.mint,
+  proposing: Cardano.RedeemerPurpose.propose,
+  reward: Cardano.RedeemerPurpose.withdrawal,
+  spend: Cardano.RedeemerPurpose.spend,
+  voting: Cardano.RedeemerPurpose.vote
+};
+
+const mapRedeemerPurpose = (purpose: RedeemerModel['purpose']): Cardano.RedeemerPurpose =>
+  redeemerPurposeMap[purpose] ||
+  (() => {
+    throw new NotImplementedError(`Failed to map redeemer "purpose": ${purpose}`);
```

Comment:
Nice implementation of error throwing in a body-less function! 🧙‍♂️ I'll re use it!